build/common/installer/scripts/tomlparser-common-agent-config.rb

```diff
@@ -62,12 +63,25 @@ def populateSettingValuesFromConfigMap(parsedConfig)
   end
 end
 
+def getKubernetesMetadataCacheTTL(parsedConfig)
+  begin
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil?
+      k8s_metadata_config = parsedConfig[:agent_settings][:k8s_metadata_config]
+      if !k8s_metadata_config.nil? && !k8s_metadata_config[:kube_meta_cache_ttl].nil?
+        @logEnableKubernetesMetadataCacheTTL = k8s_metadata_config[:kube_meta_cache_ttl]
+        puts "Using config map value: kube_meta_cache_ttl = #{@logEnableKubernetesMetadataCacheTTL}"
+      end
+    end
+  end
+end
+
 @configSchemaVersion = ENV["AZMON_AGENT_CFG_SCHEMA_VERSION"]
 puts "****************Start Config Processing********************"
 if !@configSchemaVersion.nil? && !@configSchemaVersion.empty? && @configSchemaVersion.strip.casecmp("v1") == 0 #note v1 is the only supported schema version , so hardcoding it
   configMapSettings = parseConfigMap
   if !configMapSettings.nil?
     populateSettingValuesFromConfigMap(configMapSettings)
+    getKubernetesMetadataCacheTTL(configMapSettings)
```

Comment:
you dont need this and populateSettingValuesFromConfigMap should take care of this.

Reply:
the getKubernetesMetadataCacheTTLSeconds is a separate function